jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -1202,6 +1203,23 @@ public void call(Invocable.Callable callable, Request request) throws Exception
             }
         }
 
+        public <T> boolean test(Predicate<T> predicate, T t, Request request)
```

Comment:
This looks unused. Should we keep it nevertheless?

Reply:
@lorban  It has been used before, but now that we don't have listening predicates.....  oh but wait yes we do.  This should be used by the context addIdleTimeoutListener wrapper that is not implemented!!!! stand by.....

Reply:
@lorban I've added the context wrapper so idle timeouts are callback in the context.  I've also added a test to check that.  Please re-review... at least that last commit.